crud/compare/comparators.lua

```diff
@@ -152,9 +153,13 @@ function comparators.gen_tuples_comparator(cmp_operator, key_parts, field_names,
     )
     local keys_comparator = comparators.gen_func(cmp_operator, updated_key_parts)
 
-    return function(lhs, rhs)
-        local lhs_key = utils.extract_key(lhs, updated_key_parts)
-        local rhs_key = utils.extract_key(rhs, updated_key_parts)
+    return function(lhs, rhs, skip_extract)
+        if skip_extract == true then
+            return keys_comparator(lhs, rhs)
+        end
+
+        local lhs_key = utils.extract_jsonpath_keys(lhs, updated_key_parts)
+        local rhs_key = utils.extract_jsonpath_keys(rhs, updated_key_parts)
```

Comment:
Do you really need it? This code should be executed only at versions where merger is not supported. But such versions usually don't support json-path indexes yet.